### PR TITLE
Cfy 2727 docker pre installed

### PR DESCRIPTION
**Pull request by adamlavie**

### CODE REVIEW

File: cosmo_tester/test_suites/test_manager_recovery/manager_recovery_base.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            'port': 22,
+            'user': self.env.management_user_name,
+            'key_filename': util.get_actual_keypath(
+                self.env,
+                self.env.management_key_path
+            ),
+            'connection_attempts': 5
+        }
+
+    def get_blueprint_inputs(self):
```
Comment (adamlavie) [inline cosmo_tester/test_suites/test_manager_recovery/manager_recovery_base.py]: not required

File: cosmo_tester/test_suites/test_manager_recovery/manager_recovery_docker_installed_test.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# cloudify-cosmo-system-tests3 region a
+UBUNTU_DOCKER_IMAGE_ID = 'b3322ff7-5e72-4459-b164-bdb800848289'
+
+
+# This test can only run on a specific hp tenant
+# that contains an ubuntu image running docker.
+class ManagerRecoveryWithDockerTest(BaseManagerRecoveryTest):
+
+    __test__ = True
+
```
Comment (adamlavie) [inline cosmo_tester/test_suites/test_manager_recovery/manager_recovery_docker_installed_test.py]: get rid of this

File: cosmo_tester/framework/cfy_helper.py [hunk 3]
```diff
@@ -69,9 +69,9 @@ def bootstrap(self,
                 task_retries=task_retries,
                 verbose=verbose).wait()
 
-    def recover(self):
+    def recover(self, task_retries):
```
Comment (iliapolo) [inline cosmo_tester/framework/cfy_helper.py]: add a default value here

File: cosmo_tester/test_suites/test_manager_recovery/manager_recovery_test.py [hunk 4]
```diff
@@ -13,106 +13,10 @@
...
-        self.cfy.bootstrap(blueprint_path=self.env._manager_blueprint_path,
-                           inputs_file=self.env.cloudify_config_path,
-                           task_retries=5,
-                           install_plugins=self.env.install_plugins)
-
-        # override the client instance to use the correct ip
-        self.client = CloudifyClient(self.cfy.get_management_ip())
-
-        self.addCleanup(self.cfy.teardown)
+    def test(self):
```
Comment (iliapolo) [inline cosmo_tester/test_suites/test_manager_recovery/manager_recovery_test.py]: give the method a proper name

File: cosmo_tester/test_suites/test_manager_recovery/manager_recovery_docker_installed_test.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+# cloudify-cosmo-system-tests3 region a
+UBUNTU_DOCKER_IMAGE_ID = 'b3322ff7-5e72-4459-b164-bdb800848289'
+
+
+# This test can only run on a specific hp tenant
+# that contains an ubuntu image running docker.
+class ManagerRecoveryWithDockerTest(BaseManagerRecoveryTest):
+
+    def test(self):
```
Comment (iliapolo) [inline cosmo_tester/test_suites/test_manager_recovery/manager_recovery_docker_installed_test.py]: method name

**Pull request merged**